src/pybind/mgr/cephadm/module.py

```diff
@@ -1643,8 +1643,19 @@ def _client_keyring_rm(
         self._kick_serve_loop()
         return HandleCommandResult()
 
-    def _get_container_image(self, daemon_name: str, use_current_daemon_image: bool = False) -> Optional[str]:
-        daemon_type = daemon_name.split('.', 1)[0]  # type: ignore
+    def get_container_image(
+        self,
+        daemon_name: str,
+        use_current_daemon_image: bool = False,
+        force_ceph_image: bool = False,
+    ) -> Optional[str]:
+        """Return an image for the given daemon_name.
+        If `use_current_daemon_image` is set the function will try to re-use
+        the image for an existing service.
+        If force_ceph_image is true the daemon_name will be ignored and the
+        main ceph container image will be returned.
+        """
+        daemon_type = daemon_name.split('.', 1)[0]
```

Comment:
The fact you're able to just remove the `# type: ignore` makes me wonder why it was there to begin with, but as long as mypy is happy I guess we're good

Reply:
Generally if I touch a piece of code with one of those I remove them and if the tests continue to pass I leave them out, otherwise I put it back. Often times, I think people put them in to deal with older versions of python/mypy that seemed quirky or complex and they didn't want to debug it.